src/main/java/vazkii/botania/common/block/subtile/generating/SubTileRafflowsia.java

```diff
@@ -74,21 +116,26 @@ public void tickFlower() {
 	public void writeToPacketNBT(CompoundNBT cmp) {
 		super.writeToPacketNBT(cmp);
 
-		if (lastFlower != null) {
-			cmp.putString(TAG_LAST_FLOWER, Registry.BLOCK.getKey(lastFlower).toString());
+		ListNBT flowerList = new ListNBT();
+		for (Block flower : lastFlowers) {
+			flowerList.add(StringNBT.valueOf(flower.getRegistryName().toString()));
 		}
-		cmp.putInt(TAG_LAST_FLOWER_TIMES, lastFlowerTimes);
+		cmp.put(TAG_LAST_FLOWERS, flowerList);
+		cmp.putInt(TAG_LAST_FLOWER_TIMES, lastFlowerCount);
+		cmp.putInt(TAG_STREAK_LENGTH, streakLength);
 	}
 
 	@Override
 	public void readFromPacketNBT(CompoundNBT cmp) {
 		super.readFromPacketNBT(cmp);
 
-		ResourceLocation id = ResourceLocation.tryCreate(cmp.getString(TAG_LAST_FLOWER));
-		if (id != null) {
-			lastFlower = Registry.BLOCK.getValue(id).orElse(null);
+		lastFlowers.clear();
+		ListNBT flowerList = cmp.getList(TAG_LAST_FLOWERS, Constants.NBT.TAG_STRING);
+		for (int i = 0; i < flowerList.size(); i++) {
+			lastFlowers.add(Registry.BLOCK.getOrDefault(ResourceLocation.tryCreate(flowerList.getString(i))));
```

Comment:
Should the entry be removed from the list if the id is missing instead of inserting an air block into the list? (getOrDefault returns air if the passed ID doesnt exist)

Reply:
It should be totally benign